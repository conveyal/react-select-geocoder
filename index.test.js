/* globals describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import nock from 'nock'
import React from 'react'

import Geocoder from './index'
const mockAutocompleteResult = require('./mock-autocomplete-result.json')

// add a div to jsdom for enzyme to mount to
const div = document.createElement('div')
div.id = 'test'
document.body.appendChild(div)

function timeoutPromise (ms) {
  return new Promise((resolve, reject) => {
    setTimeout(resolve, ms)
  })
}

describe('Geocoder', () => {
  it('should render', () => {
    const tree = mount(
      <Geocoder
        apiKey='test'
        />
    , {
      attachTo: document.getElementById('test')
    })
    expect(mountToJson(tree)).toMatchSnapshot()
  })

  it('should process input change', async () => {
    nock('https://search.mapzen.com/')
      .get(/v1\/autocomplete/)
      .reply(200, mockAutocompleteResult)

    const tree = mount(
      <Geocoder
        apiKey='test'
        />
    , {
      attachTo: document.getElementById('test')
    })

    let calculatedOptions

    tree.find('Async').props().loadOptions('123 main', (error, result) => {
      expect(error).toBeFalsy()
      calculatedOptions = result
    })

    // wait for query to mapzen
    await timeoutPromise(1000)

    expect(calculatedOptions.options).toHaveLength(10)
  })
})
