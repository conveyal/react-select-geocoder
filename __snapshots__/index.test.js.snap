exports[`Geocoder should render 1`] = `
<Geocoder
  apiKey="test"
  featureToLabel={[Function]}
  featureToValue={[Function]}
  findingLocationText="Locating you..."
  reverseSearch={[Function]}
  search={[Function]}
  useLocationText="Use Current Location">
  <Async
    apiKey="test"
    autoBlur={true}
    autoload={false}
    cache={false}
    featureToLabel={[Function]}
    featureToValue={[Function]}
    filterOptions={false}
    findingLocationText="Locating you..."
    ignoreAccents={false}
    ignoreCase={false}
    loadOptions={[Function]}
    loadingPlaceholder="Loading..."
    minimumInput={3}
    onChange={[Function]}
    options={Array []}
    reverseSearch={[Function]}
    search={[Function]}
    searchPromptText="Type to search"
    useLocationText="Use Current Location"
    value={null}>
    <Select
      addLabelText="Add \"{label}\"?"
      apiKey="test"
      arrowRenderer={[Function]}
      autoBlur={true}
      autoload={false}
      autosize={true}
      backspaceRemoves={true}
      backspaceToRemoveMessage="Press backspace to remove {label}"
      cache={false}
      clearAllText="Clear all"
      clearRenderer={[Function]}
      clearValueText="Clear value"
      clearable={true}
      deleteRemoves={true}
      delimiter=","
      disabled={false}
      escapeClearsValue={true}
      featureToLabel={[Function]}
      featureToValue={[Function]}
      filterOptions={false}
      findingLocationText="Locating you..."
      ignoreAccents={false}
      ignoreCase={false}
      inputProps={Object {}}
      isLoading={false}
      joinValues={false}
      labelKey="label"
      loadOptions={[Function]}
      loadingPlaceholder="Loading..."
      matchPos="any"
      matchProp="any"
      menuBuffer={0}
      menuRenderer={[Function]}
      minimumInput={3}
      multi={false}
      noResultsText="Type to search"
      onBlurResetsInput={true}
      onChange={[Function]}
      onCloseResetsInput={true}
      onInputChange={[Function]}
      openAfterFocus={false}
      optionComponent={[Function]}
      options={Array []}
      pageSize={5}
      placeholder="Select..."
      required={false}
      reverseSearch={[Function]}
      scrollMenuIntoView={true}
      search={[Function]}
      searchPromptText="Type to search"
      searchable={true}
      simpleValue={false}
      tabSelectsValue={true}
      useLocationText="Use Current Location"
      value={null}
      valueComponent={[Function]}
      valueKey="value">
      <div
        className="Select Select--single is-searchable">
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}>
          <span
            className="Select-multi-value-wrapper"
            id="react-select-2--value">
            <div
              className="Select-placeholder">
              Select...
            </div>
            <AutosizeInput
              aria-activedescendant="react-select-2--value"
              aria-expanded="false"
              aria-haspopup="false"
              aria-owns=""
              className="Select-input"
              minWidth="5"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              value="">
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }>
                <input
                  aria-activedescendant="react-select-2--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value="" />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  } />
              </div>
            </AutosizeInput>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}>
            <span
              className="Select-arrow"
              onMouseDown={[Function]} />
          </span>
        </div>
      </div>
    </Select>
  </Async>
</Geocoder>
`;
